interfaces/api/pipeline/IImageSegmentationPipeline.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * @copyright Copyright (c) 2025 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SOLAR_IMAGESEGMENTATIONPIPELINE_H
+#define SOLAR_IMAGESEGMENTATIONPIPELINE_H
+
+#include "api/pipeline/IPipeline.h"
+#include "datastructure/Image.h"
+#include "datastructure/Mask2DCollection.h"
+#include <xpcf/core/helpers.h>
+
+namespace SolAR {
+using namespace datastructure;
+namespace api {
+namespace pipeline {
+
+/**
+ * @enum class ImageSegmentationStatus
+ */
+enum class ImageSegmentationStatus {
+    UNINITIALIZED = 0,    // processing not initialized
+    INITIALIZED = 1,      // processing correctly initialized, but not started
+    IN_PROGRESS = 2,      // processing in progress
+    COMPLETED = 3,        // processing completed
+    ABORTED = 4           // processing aborted before completion
+};
+
+/// @brief image segmentation status to string
+static const std::map<ImageSegmentationStatus, std::string> imageSegmentationStatusToString = {
+    {ImageSegmentationStatus::UNINITIALIZED, "UNINITIALIZED"},
+    {ImageSegmentationStatus::INITIALIZED, "INITIALIZED"},
+    {ImageSegmentationStatus::IN_PROGRESS, "IN_PROGRESS"},
+    {ImageSegmentationStatus::COMPLETED, "COMPLETED"},
+    {ImageSegmentationStatus::ABORTED, "ABORTED"}
+};
+
+/**
+ * @class IImageSegmentationPipeline
+ * @brief <B>Defines an image segmentation pipeline.</B>
+ * <TT>UUID: 0a897dee-74f1-42de-a6c1-f7855e0f0bbb</TT>
+ *
+ * This class provides the interface to define an image segmentation pipeline.
+ */
+
+class XPCF_CLIENTUUID("2215b6ef-e6fa-455c-84c6-820d95630eb5") XPCF_SERVERUUID("40aede65-0170-4384-9623-6a9d320ae56d")
+    XPCF_GRPC_CLIENT_RECV_SIZE("-1") XPCF_GRPC_CLIENT_SEND_SIZE("-1")
+    IImageSegmentationPipeline : virtual public IPipeline {
+public:
+    /// @brief default constructor
+    IImageSegmentationPipeline() = default;
+
+    /// @brief default destructor
+    virtual ~IImageSegmentationPipeline() = default;
+
+    /// @brief segmentation request from an input image
+    /// @param[in] image pointer to image
+    /// @return FrameworkReturnCode::_SUCCESS (segmentation succeeded) or FrameworkReturnCode::_ERROR_ (segmentation failed)
+    virtual FrameworkReturnCode segmentationRequest(SRef<Image> image) = 0;
```

Comment:
nit: 

I know it's been done elsewhere, but not a fan of this name:
* usually a verb is used for method (action). Here it sounds like an object
* request is redundant. If we call a method, we make a request to the object to perform an action.

It could be for example: segment(), run/performSegmentation(), or something else.

If we want to be concise, it could even be just `run()`-, because calling `IImageSegmentationPipeline::run()` it is clear enough that run() will perform an image segmentation.

Also, I'm not sure, but enum ImageSegmentationStatut could be defined inside the class, and simply be called Status. The method getStatus() is not called getImageSegmentationStatus().

Just some thoughts, you decide ;)